WordPress/gutenberg
# `FormTokenField`: handle `disabled` prop on internal `Button`

replaces #23713

## What?
Pass the `disabled` prop to the `Button` component on selected tokesn

## Why?
When `FormTokenField` is `disabled`, the buttons don't remove tokens because the click isn't handled. The buttons are still focusable though, which isn't ideal.

## How?
prop forwarding

## Testing Instructions
1. Launch Storybook and test the Default `FormTokenField` story
2. Select a couple of options
3. In the Controls panel, set `Disabled` to `true`
4. Confirm that the tokens you've added cannot be removed, and clicking on the **x** button doesn't apply focus.
